src/build.py

```diff
@@ -1323,144 +1327,199 @@ def Test(self):
 
 def TestBare():
   # Compile
-  CompileLLVMTorture('s', TORTURE_S_OUT_DIR)
-  CompileLLVMTorture('o', TORTURE_O_OUT_DIR)
+  for opt in BARE_TEST_OPT_FLAGS:
+    CompileLLVMTorture('s', os.path.join(TORTURE_S_OUT_DIR, opt), opt)
+  for opt in BARE_TEST_OPT_FLAGS:
+    CompileLLVMTorture('o', os.path.join(TORTURE_O_OUT_DIR, opt), opt)
 
   # Link/Assemble
+  lld_musl_out = {}
```

Comment:
Hm, this feels squirrely to me, but I think I'm just being weird.

This is a very natural extension of precomputing the directory name. I much prefer it to reusing the `os.path.join` logic.

I think my unease comes from how this introduces state that gets computed as the program runs, where we
1. compute and store -O0 dir
1. link -O0 files
1. compute and store -O2 dir
1. link -O2 files
1. run -O0 files
1. run -O2 files
and that interleaving of storing the dir with linking seems off. In theory we could add another `for opt in BARE_TEST_OPT_FLAGS` to precompute the dirs, then extract that into a function giving us something like `lld_musl_out = PrecomputeOutDirs(os.path.join(WORK_DIR, 'torture-lld-musl'), BARE_TEST_OPT_FLAGS)`

I also like that because it avoids repeating this pattern for the other cases, but, again, it might be total overkill.

Reply:
Yeah, I wasn't super satisfied with the flow either actually. So the original flow had this kind of chain:
```
s2wasm_out = LinkLLVMTorture('s2wasm', ...)
wast2wasm_out = AssembleLLVMTorture(..., indir=s2wasm_out, ...)
ExecuteLLVMTorture(..., indir=wast2wasm_out, ...)
```

But as we introduce a compilation flag loop for each of them, this chain does not work anymore, which was the reason I introduced this map thing, which wasn't super tidy either.

And the other thing is, all those ***LLVMTorture functions had different ways of getting `indir` and `outdir` - some of them compute them within the function while others take them as arguments. Some function only take indir as an argument and others take both `indir`/`outdir`, and all of this was confusing, so I made all of torture functions take both `indir`/`outdir` as arguments (except for `CompileLLVMTorture` and `CompileLLVMTortureBinaryen`, which takes c codes form gcc test suite directory). Then I couldn't help but wonder, why all these functions take `outdir` as their parameter and return the very same thing as a result? So I decided to delete all the return values from all torture functions and compute all directories only from outside.

So far was the not-very-interesting flow of my thoughts. Anyway I created `GetTortureDir` function and made all other functions inquire this function to get torture directories. PTAL.